homeassistant/components/homekit_controller/connection.py

```diff
@@ -135,20 +135,33 @@ def async_set_unavailable(self):
         self.accessories = cache["accessories"]
         self.config_num = cache["config_num"]
 
-        # Ensure the Pairing object has access to the latest version of the
-        # entity map.
+        self._polling_interval_remover = async_track_time_interval(
+            self.hass, self.async_update, DEFAULT_SCAN_INTERVAL
+        )
+
+        self.hass.async_create_task(self.async_process_entity_map())
+
+        return True
+
+    async def async_process_entity_map(self):
+        """
+        Process the entity map and load any platforms or entities that need adding.
+
+        This is idempotent and will be called at startup and when we detect metadata changes
+        via the c# counter on the zeroconf record.
+        """
+        # Ensure the Pairing object has access to the latest version of the entity map. This
+        # is especially important for BLE, as the Pairing instance relies on the entity map
+        # to map aid/iid to GATT characteristics. So push it to there as well.
+
         self.pairing.pairing_data["accessories"] = self.accessories
 
-        self.async_load_platforms()
+        await self.async_load_platforms()
 
         self.add_entities()
 
         await self.async_update()
 
-        self._polling_interval_remover = async_track_time_interval(
-            self.hass, self.async_update, DEFAULT_SCAN_INTERVAL
-        )
-
         return True
```

Comment:
Nothing is checking this return value.

Reply:
Good point - i have removed it in my latest commit.